tests/PPC/TestLifting.cpp

```diff
@@ -0,0 +1,801 @@
+           return (st.gpr.r8.qword);
+         }},
+        {"r9",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return (st.gpr.r9.qword);
+         }},
+        {"r10",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.gpr.r10.qword;
+         }},
+        {"r11",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.gpr.r11.qword;
+         }},
+        {"r12",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.gpr.r12.qword;
+         }},
+        {"cr",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.iar.cr.qword;
+         }},
+        {"cr0",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr0;
+         }},
+        {"cr1",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr1;
+         }},
+        {"cr2",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr2;
+         }},
+        {"cr3",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr3;
+         }},
+        {"cr4",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr4;
+         }},
+        {"cr5",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr5;
+         }},
+        {"cr6",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr6;
+         }},
+        {"cr7",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.cr_flags.cr7;
+         }},
+        {"lr",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.iar.lr.qword;
+         }},
+        {"ctr",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.iar.ctr.qword;
+         }},
+        {"xer",
+         [](PPCState &st) -> std::reference_wrapper<uint64_t> {
+           return st.iar.xer.qword;
+         }},
+        {"xer_so",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.xer_flags.so;
+         }},
+        {"xer_ov",
+         [](PPCState &st) -> std::reference_wrapper<uint8_t> {
+           return st.xer_flags.ov;
+         }},
+};
+
+std::optional<remill::Instruction>
+GetFlows(std::string_view bytes, uint64_t address, uint64_t vle_val) {
+
+  llvm::LLVMContext context;
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(15, 0)
+  context.enableOpaquePointers();
+#endif
+  auto arch = remill::Arch::Build(&context, remill::OSName::kOSLinux,
+                                  remill::ArchName::kArchPPC);
+  auto sems = remill::LoadArchSemantics(arch.get());
+
+
+  remill::DecodingContext dec_context;
+  dec_context.UpdateContextReg(std::string("VLEReg"), vle_val);
```

Comment:
This is currently hard coded with PPC but it would be nice if this could be shared among tests for other arches